File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not len(s):return True
        schar = 0;found = 0
        for i in t:
            if schar >= len(s):break
            if i == s[schar]:
                schar += 1; found += 1
        return found == len(s)

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        char[] sc = s.toCharArray();
        char[] tc = t.toCharArray();
        int i = 0, j = 0;
        while (i < sc.length && j < tc.length) {
            if (sc[i] == tc[j]) {
                i++;
            }
            j++;
        }

        return i == sc.length;
    }
}